### PR TITLE
ADEN-5242 Ooyala tracker for featured video

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 1]
```diff
@@ -6,8 +6,19 @@ require([
 	'wikia.abTest',
 	'wikia.articleVideo.videoFeedbackBox',
 	require.optional('ext.wikia.adEngine.adContext'),
+	require.optional('ext.wikia.adEngine.video.player.ooyala.ooyalaTracker'),
 	require.optional('ext.wikia.adEngine.video.vastUrlBuilder')
-], function (window, onScroll, tracker, OoyalaPlayer, abTest, VideoFeedbackBox, adContext, vastUrlBuilder) {
+], function (
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js:11]: MAJOR Function has 9 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 2]
```diff
@@ -37,9 +52,13 @@ require([
...
 				vastUrl;
 
+			if (playerTracker) {
+				playerTracker.track(playerTrackerParams, 'init');
+			}
+
 			if (vastUrlBuilder && adContext && adContext.getContext().opts.showAds) {
 				vastUrl = vastUrlBuilder.build(640/480, {
-					pos: 'FEATURED_VIDEO',
+					pos: playerTrackerParams.slotName,
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: I don't feel great when we're using `playerTrackerParams` for ads targeting. Maybe we should have a separate variable for ad slot name that we can use in both places?

File: extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	 * @param {string} params.adProduct
+	 * @param {string} [params.slotName]
+	 * @param {string} eventName
+	 */
+	function track(params, eventName) {
+		playerTracker.track(params, playerName, eventName);
+	}
+
+	/**
+	 * @param {object} player created by porvataPlayerFactory
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js]: C&P? Is it accurate?

File: extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+			'adsError': 'error',
+			'adsPlayed': 'completed',
+			'willPlayAds': 'loaded',
+			'playerCreated': 'ready',
+			'willPauseAds': 'paused',
+			'willResumeAds': 'resumed',
+			'willPlaySingleAd': 'started',
+
+			'willPlay': 'content_started',
+			'played': 'content_completed'
```
Comment (kvas-damian) [inline extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js]: what do you think about using contants provided by Ooyala?
eg. `OO.EVENTS.PLAYED`

**Review by kvas-damian — APPROVED** [review]

File: extensions/wikia/AdEngine/js/video/player/playerTracker.js [hunk 5]
```diff
@@ -3,23 +3,23 @@ define('ext.wikia.adEngine.video.player.playerTracker', [
...
 	'ext.wikia.adEngine.adTracker',
-	'ext.wikia.adEngine.lookup.prebid.bidHelper',
 	'ext.wikia.adEngine.slot.slotTargeting',
 	'wikia.browserDetect',
 	'wikia.geo',
 	'wikia.log',
 	'wikia.window',
+	require.optional('ext.wikia.adEngine.lookup.prebid.bidHelper'),
 	require.optional('ext.wikia.adEngine.video.player.porvata.floater')
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/video/player/playerTracker.js:13]: MAJOR Function has 10 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/ae526f188f19165347433939ebb6339dd5de29ce/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has a complexity of 21 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/ae526f188f19165347433939ebb6339dd5de29ce/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [playerTracker.js#L90](https://github.com/Wikia/app/blob/ae526f188f19165347433939ebb6339dd5de29ce/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L90): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/dddb70693540e638db48985940f90700705c5139/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has a complexity of 21 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/dddb70693540e638db48985940f90700705c5139/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [playerTracker.js#L90](https://github.com/Wikia/app/blob/dddb70693540e638db48985940f90700705c5139/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L90): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	 * @param {string} params.adProduct
+	 * @param {string} [params.slotName]
+	 * @param {string} eventName
+	 */
+	function track(params, eventName) {
+		playerTracker.track(params, playerName, eventName);
+	}
+
+	/**
+	 * @param {object} player
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js]: What do you think of describing more `player` object at least describing `mb` property with a link to Ooyala docs?

File: extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+			WILL_PAUSE_ADS: 'paused',
+			WILL_RESUME_ADS: 'resumed',
+			WILL_PLAY_SINGLE_AD: 'started',
+
+			WILL_PLAY: 'content_started',
+			PLAYED: 'content_completed'
+		};
+
+	/**
+	 * @param {object} params
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/player/ooyala/ooyalaTracker.js]: What do you think of providing minimal docs? Like:
```js
/**
 * @param {object} params a configuration object passed from DFP player creative
 * @param {string} params.adProduct a name of ad product passed to DW
 * @param {string} [params.slotName] ad slot name which will be stored in DW
 * @param {string} eventName name of an event passed to DW
 */
```

File: extensions/wikia/AdEngine/js/video/player/playerTracker.js [hunk 8]
```diff
@@ -3,23 +3,23 @@ define('ext.wikia.adEngine.video.player.playerTracker', [
...
 	adContext,
 	pageLevel,
 	adTracker,
-	bidHelper,
 	slotTargeting,
 	browserDetect,
 	geo,
 	log,
 	win,
+	bidHelper,
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/video/player/playerTracker.js:22]: What's the reason for moving it here?A race condition when loading `allinone=0`?

Comment (nandy-andy) [reply extensions/wikia/AdEngine/js/video/player/playerTracker.js:22]: Oh, it's optional. It's fine. Thanks :bowing_man:

Comment (fraszczakszymon) [reply extensions/wikia/AdEngine/js/video/player/playerTracker.js:22]: Discussed offline

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/734d8084d840fae53d17f559ead60fa1edcaa7ca/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has a complexity of 21 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/734d8084d840fae53d17f559ead60fa1edcaa7ca/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [playerTracker.js#L90](https://github.com/Wikia/app/blob/734d8084d840fae53d17f559ead60fa1edcaa7ca/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L90): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by nandy-andy — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/ab64e7a0c1c541171454d3555f66f76334b36ffb/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has a complexity of 21 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/ab64e7a0c1c541171454d3555f66f76334b36ffb/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [playerTracker.js#L90](https://github.com/Wikia/app/blob/ab64e7a0c1c541171454d3555f66f76334b36ffb/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L90): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/159ef09b5e51639242094d25c43503320b455b77/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has a complexity of 21 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [playerTracker.js#L38](https://github.com/Wikia/app/blob/159ef09b5e51639242094d25c43503320b455b77/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L38): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [playerTracker.js#L90](https://github.com/Wikia/app/blob/159ef09b5e51639242094d25c43503320b455b77/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L90): Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by dianafa — APPROVED** [review]

**Pull request merged**